shares/share_builder.go

```diff
@@ -168,6 +168,17 @@ func (b *Builder) WriteSequenceLen(sequenceLen uint32) error {
 	return nil
 }
 
+// WriteSigner writes the signer's information to the share.
+func (b *Builder) WriteSigner(signer []byte) {
+	// only write the signer if it is the first share and the share version is 1
+	if b == nil || !b.isFirstShare || b.shareVersion != ShareVersionOne {
+		return
+	}
+	// NOTE: we don't check whether previous data has already been expected
```

Comment:
[nit]
I guess this note should also be written in the function docs so that users can see it when they're using it, instead of having to read the code to know

Reply:
I think instead I want to make an assertion about the size of the `rawShareData` before making the append to know whether the other writes have happened. Personally, I'm not sure about the entire builder struct (I think it should at least be private)